packages/server-wallet/src/wallet/__test__/sign-state.test.ts

```diff
@@ -0,0 +1,58 @@
+import Objection from 'objection';
+
+import {Store} from '../store';
+import {signState} from '../../protocols/actions';
+import {channel} from '../../models/__test__/fixtures/channel';
+import {seed} from '../../db/seeds/1_signing_wallet_seeds';
+import knex from '../../db/connection';
+import {Channel} from '../../models/channel';
+
+import {stateWithHashSignedBy} from './fixtures/states';
+import {bob} from './fixtures/signingWallets';
+
+let tx: Objection.Transaction;
+beforeEach(async () => {
+  // Make sure alice's PK is in the DB
+  await seed(knex);
```

Comment:
One thing I noticed is that since we re-use these fixtures and clear the DB on every test, it makes it hard to parallelize the jest tests. Do we want to parallelize the jest tests? Could we do something so that instead of running this global seed there is some per-test unique seed? Out of scope for the PR

Reply:
One large advantage of writing protocols and "handlers" as pure functions that decide what to do, but don't do it, is that it's trivial to run those tests in parallel.

So, we can have one test job of these protocol level decisions, which test the complicated logic, and run in parallel.

We can have another test job that tests the integration of those protocol decisions into our database store, which runs in serial.

If we are unhappy running those tests in serial, we can explore some techniques for running them in parallel, like executing each test case inside a transaction.